src/petals/models/bloom/model.py

```diff
@@ -39,16 +39,15 @@ def __init__(self, config: DistributedBloomConfig, *, dht: Optional[hivemind.DHT
     def forward(
         self,
         input_ids: Optional[torch.LongTensor] = None,
-        inputs_embeds: Optional[torch.Tensor] = None,
+        past_key_values: Optional[RemotePastKeyValues] = None,
```

Comment:
Not a request, but a sanity check: when we support prefix_tuning, this logic will need to change. Superficially, it should be simple to support both RemotePastKeyValues and nested pytorch tensors as past_key_values.
__Can you think of some caveats?__

For instance, how do we pass `past_key_values` tensors from prefix_tuning during .generate , when inferencing a prefix-tuned model?

Reply:
Prefix-tuning past kvs are only used on the 1st generation step, while mocked `RemotePastKeyValues` here are only used for 2nd step and later to indicate that we have **some** past kvs (it's not important which ones for HF's `.generate()`). So I don't think these things contradict each other or could introduce any conflicts.